tests/ws.c

```diff
@@ -29,15 +30,58 @@
 
 #include "testutil.h"
 
+#define SOCKET_ADDRESS  "ws://127.0.0.1:5555"
+
 /*  Basic tests for WebSocket transport. */
 
+/*  test_text() verifies that we drop messages properly when sending invalid
+    UTF-8, but not when we send valid data. */
+void test_text() {
+
+    int sb;
+    int sc;
+    int opt;
+    uint8_t bad[20];
+
+    /*  Negative testing... bad UTF-8 data for text. */
+    sb = test_socket (AF_SP, NN_PAIR);
+    sc = test_socket (AF_SP, NN_PAIR);
+
+    /*  Wait for connects to establish. */
+    nn_sleep (200);
+
+    opt = NN_WS_MSG_TYPE_TEXT;
+    test_setsockopt(sb, NN_WS, NN_WS_MSG_TYPE, &opt, sizeof (opt));
+    opt = NN_WS_MSG_TYPE_TEXT;
+    test_setsockopt(sc, NN_WS, NN_WS_MSG_TYPE, &opt, sizeof (opt));
+    opt = 500;
+    test_setsockopt(sb, NN_SOL_SOCKET, NN_RCVTIMEO, &opt, sizeof (opt));
+
+    test_bind (sb, SOCKET_ADDRESS);
+    test_connect (sc, SOCKET_ADDRESS);
+
+    test_send (sc, "GOOD");
+    test_recv (sb, "GOOD");
+
+    /*  and the bad ... */
+    strcpy((char *)bad, "BAD.");
+    bad[2] = (char)0xDD;
+    test_send (sc, (char *)bad);
+
+    /*  Make sure we dropped the frame.  Apparently it comes back as EAGAIN
+        rather than ETIMEDOUT.  I think this possibly a doc bug. */
```

Comment:
The right place to report this, if ever we could, would be at the *sender side*.  Arguably in *that* case we could actually capture this and report it back to the sender.  Even that might not be very satisfactory in the face of e.g. device frameworks and proxies, that are just relaying data.

This is all part of the design for nanomsg -- it always just works (except when it doesnt'), and never reports errors back to the application.  (Unfortunately it *does* crash/core on assertion failures far more often than I'd like, but that's a separate rant.)